src/interval.jl

```diff
@@ -168,10 +175,7 @@ Base.isless(a::T, b::AbstractInterval{T}) where T = a < LeftEndpoint(b)
 ##### SET OPERATIONS #####
 
 function Base.isempty(i::Interval)
-    return (
-        first(i) > last(i) ||   # Shouldn't be possible, but check anyway.
-        (first(i) == last(i) && inclusivity(i) != Inclusivity(true, true))
-    )
+    return first(i) == last(i) && inclusivity(i) != Inclusivity(true, true)
```

Comment:
Once https://github.com/invenia/Intervals.jl/pull/7 is merged we can change this to:
```julia
LeftEndpoint(i) == RightEndpoint(i)
```

Reply:
Should I make this change in #7?

Reply:
No, I don't think that will work. You'll still need to verify that at least one of the endpoints is exclusive in order for the interval to be empty:

```
isempty(Interval(10, 10, false, false)) == true
isempty(Interval(10, 10, true, true)) == false    # would be true in your example
isempty(Interval(10, 10, true, false)) == true    # would be false in your example
isempty(Interval(10, 10, false, true)) == true    # would be false in your example
```

Reply:
My original comment about the implementation is definitely wrong. As it turns out it would be:
```julia
isempty(i::AbstractInterval) = LeftEndpoint(i) > RightEndpoint(i)
```

Reply:
I'm not super familiar with the way the endpoint inclusivity checks are done, but so long as the existing tests pass, this should be fine.

Reply:
Since we're not using the new `==` in #7 we could actually do this implementation here. I'll just make a new PR for this change to keep everything separated

Reply:
https://github.com/invenia/Intervals.jl/pull/10

Reply:
I'll merge that in next. Thanks.